compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -35,11 +35,8 @@ fn inline(cx: &CodegenCx<'ll, '_>, val: &'ll Value, inline: InlineAttr) {
                 Attribute::NoInline.apply_llfn(Function, val);
             }
         }
-        None => {
-            Attribute::InlineHint.unapply_llfn(Function, val);
-            Attribute::AlwaysInline.unapply_llfn(Function, val);
-            Attribute::NoInline.unapply_llfn(Function, val);
-        }
+        None if requires_inline => Attribute::InlineHint.apply_llfn(Function, val),
```

Comment:
What's the intended behaviour here when `requires_inline == true` and `inline != None`?

Reply:
If its not something that can happen, maybe `bug!()` it?

Reply:
Requires inline instances are mostly auto-generated. Closures are the exception, so it is possible but unlikely. As far as what should take priority, no idea what the intent was, if it was considered at all (although the order of calls would suggest respecting user provided attributes).